spec/services/proofing/socure/id_plus/request_spec.rb

```diff
@@ -0,0 +1,241 @@
+    end
+
+    it 'includes JSON serialized body' do
+      request.send_request
+
+      expect(WebMock).to have_requested(
+        :post, 'https://example.org/api/3.0/EmailAuthScore'
+      ).with(body: request.body)
+    end
+
+    context 'when service returns HTTP 200 response' do
+      it 'method returns a Proofing::Socure::IdPlus::Response' do
+        res = request.send_request
+        expect(res).to be_a(Proofing::Socure::IdPlus::Response)
+      end
+
+      it 'response has kyc data' do
+        res = request.send_request
+        expect(res.kyc_field_validations).to be
+        expect(res.kyc_reason_codes).to be
+      end
+    end
+
+    context 'when service returns an HTTP 400 response' do
+      before do
+        stub_request(:post, 'https://example.org/api/3.0/EmailAuthScore').
+          to_return(
+            status: 400,
+            headers: {
+              'Content-Type' => 'application/json',
+            },
+            body: JSON.generate(
+              {
+                status: 'Error',
+                referenceId: 'a-big-unique-reference-id',
+                data: {
+                  parameters: ['firstName'],
+                },
+                msg: 'Request-specific error message goes here',
+              },
+            ),
+          )
+      end
+
+      it 'raises RequestError' do
+        expect do
+          request.send_request
+        end.to raise_error(
+          Proofing::Socure::IdPlus::RequestError,
+          'Request-specific error message goes here (400)',
+        )
+      end
+
+      it 'includes reference_id on RequestError' do
+        expect do
+          request.send_request
+        end.to raise_error(
+          Proofing::Socure::IdPlus::RequestError,
+        ) do |err|
+          expect(err.reference_id).to eql('a-big-unique-reference-id')
+        end
+      end
+    end
+
+    context 'when service returns an HTTP 401 reponse' do
+      before do
+        stub_request(:post, 'https://example.org/api/3.0/EmailAuthScore').
+          to_return(
+            status: 401,
+            headers: {
+              'Content-Type' => 'application/json',
+            },
+            body: JSON.generate(
+              {
+                status: 'Error',
+                referenceId: 'a-big-unique-reference-id',
+                msg: 'Request-specific error message goes here',
+              },
+            ),
+          )
+      end
+
+      it 'raises RequestError' do
+        expect do
+          request.send_request
+        end.to raise_error(
+          Proofing::Socure::IdPlus::RequestError,
+          'Request-specific error message goes here (401)',
+        )
+      end
+    end
+
+    context 'when service returns weird HTTP 500 response' do
+      before do
+        stub_request(:post, 'https://example.org/api/3.0/EmailAuthScore').
+          to_return(
+            status: 500,
+            body: 'It works!',
```

Comment:
```suggestion
            body: 'It does not work!',
```

(Only half-joking?)

Reply:
Oh, ha! I am all for Apache jokes. I somehow missed this one, but I am `200 OK` with this now.